docs/HowToPortLoadableAppsConfiguration.md

```diff
@@ -0,0 +1,229 @@
+# Porting guide for Loadable Applications configuration on TizenRT
+
+## CONTENTS
+
+- [Overview of Loadable Apps](#overview-of-loadable-apps)
+- [Assumptions and Limitations](#assumptions-and-limitations)
+- [Steps to enable existing loadable apps configuration](#steps-to-enable-existing-loadable-apps-configuration)
+- [Steps to add new loadable apps configuration](#steps-to-add-new-loadable-apps-configuration)
+
+## Overview of Loadable Apps
+- In case of Loadable Apps, in addition to a separate kernel binary, separate binaries are generated for each user space application. These binaries are stored in the flash & loaded to the RAM during runtime.
+- Binary manager is a kernel thread to manage binaries and recover the faults in the system. The fault manager integrates into binary manager as recovery thread. It will identify the faulty app and take necessary action.
+- The priority of binary manager is 203 and the binary manager loader is 200 whereas the prioritie of hpwork thread is 201. This is to ensure that hpwork must be running during the binary manager loader.
+- This ensures the Real-time threads to be prioritized even during loading of binaries. The priorities in order are as follows;
+:
+```
+	----------------------------------------
+	MODULE				PRIORITY
+	----------------------------------------
+	BINARY_MANAGER_PRIORITY		  203
+	CONFIG_SCHED_HPWORKPRIORITY	  201
+	LOADINGTHD_PRIORITY		  200
+	CONFIG_SCHED_LPWORKPRIORITY	  50
+	----------------------------------------
+```
+- Loadable apps are needed for better fault management in the system. In case any fault is detected in the system, the application could be reloaded without the need for the whole system to be rebooted (criticality & configuration dependent).
+- There is an MPU based memory isolation & protection in place for each application. These apps are called loadable as they can be
+reloaded into their memory space even after fault detection.
+- TizenRT provides 2 sample loadable apps called micom and wifi in TizenRT 3.0.
+- Apart from these, the user can also add new loadable apps as per the requirement.
+
+## Assumptions and Limitations
+
+1. Protected build is enabled in the configuration. (CONFIG_BUILD_PROTECTED=y)
```

Comment:
How about giving the link of porting guide for protected build?

Reply:
yes that would be good , i will add it